### PR TITLE
feat!: v3 changes

### PR DESCRIPTION
**Pull request by mdonnalley**
- Makes `cli` property on `execCmd` dictate which executable is used. 
  - `inherit`: use `TESTKIT_EXECUTABLE_PATH` or local executable if not set
  - `sfdx`: use `sfdx`
  - `sf`: use `sf`

- Renames `authStrategy` to `devhubAuthStrategy` and default to `NONE`
- Adds `AUTO` option to `devhubAuthStrategy` to replace v2's existing default (determine the method by which env vars are present)
- Removes `setupCommands` from `TestSession`
- Adds `scratchOrgs` option to `TestSession` and saves `AuthFields` to `TestSession.orgs`

    ```typescript
    const testSession = await TestSession.create({
      scratchOrgs: [{ edition: 'developer', config: 'config/project-scratch-def.json', setDefault: true }],
    });
    // if setDefault, then the default org will be accessible with the 'default' key. Otherwise, you can use the org's username to access its auth fields
    const authFields = testSession.orgs.get('default'); 
    execCmd(`force:source:deploy -x package.xml -u ${authFields.username}`, { ensureExitCode: 0 });
    ```

@W-11823386@

### CODE REVIEW

File: src/execCmd.ts [hunk 1]
```diff
@@ -122,21 +109,22 @@ const getExitCodeError = (cmd: string, expectedCode: number, output: ShellString
...
-  const debug = Debug('testkit:buildCmd');
+const determineExecutable = (cli: CLI = 'inherit'): string => {
+  const debug = Debug('testkit:determineExecutable');
 
   const bin =
-    env.getString('TESTKIT_EXECUTABLE_PATH') ??
-    pathJoin(process.cwd(), 'bin', process.platform === 'win32' ? 'dev.cmd' : 'dev');
+    cli === 'inherit'
+      ? env.getString('TESTKIT_EXECUTABLE_PATH') ??
+        pathJoin(process.cwd(), 'bin', process.platform === 'win32' ? 'dev.cmd' : 'dev')
```
[the remark this answers is not in the document]
Comment (mshanemc) [reply src/execCmd.ts:126]: Eric's got some numbers on this and is doing some more work during devChoice.

Not sure if you want to think about what testkit's interface would look like if I said, "how to do specify bin/run?"

**Review by mshanemc — APPROVED** [review]
non blocking comments

File: src/execCmd.ts [hunk 2]
```diff
@@ -58,33 +65,13 @@ export interface ExecCmdResult {
...
-
-const DEFAULT_EXEC_OPTIONS: ExecCmdOptions = {
-  cli: 'sfdx',
-};
-
 const buildCmdOptions = (options?: ExecCmdOptions): ExecCmdOptions => {
-  const defaults: shelljs.ExecOptions = {
-    env: Object.assign({}, process.env),
+  const defaults: ExecCmdOptions = {
+    env: { ...process.env },
```
[the remark this answers is not in the document]
Comment (mdonnalley) [reply src/execCmd.ts:70]: I'm actually not sure why this is needed. I only changed it from `Object.assign` to the spread to be more idiomatic

File: src/execCmd.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (mdonnalley) [reply src/execCmd.ts:126]: Interesting. I'm not opposed to adding an option for it, however, if consumers use `bin/run` they loose the ability to access any dev plugins they want to use

**Pull request merged**